src/polycubed/src/cubes_dump.cpp

```diff
@@ -0,0 +1,336 @@
+    for (int i = 2; i < resItem.size() - 1; i++) {
+      auto *reference = &(*item)[resItem[i]];
+      if (reference->is_array()) { /* If it is an array corresponding to a key, then it is an element of an array */
+        reference = &*(std::find_if(reference->begin(), reference->end(),
+                [resItem, i, keyValues, this](auto const& elem){return checkPresence(resItem, i, keyValues, elem);}));
+        i += keyValues.at(resItem[i]).size();
+      }
+      item = reference;
+    }
+
+    nlohmann::json toUpdate = nlohmann::json::object();
+    toUpdate[resItem[resItem.size() - 1]] = body;
+    toUpdate.insert(item->begin(), item->end());
+    item->clear();
+    item->update(toUpdate);
+  }
+}
+
+void CubesDump::UpdateCubesConfigDelete(const std::vector<std::string> &resItem,
+                                        const nlohmann::json& body,
+                                        const ListKeyValues &keys,
+                                        Rest::Resources::Endpoint::ResourceType resType) {
+
+  const std::string &serviceName = resItem[0];
+  const std::string &cubeName = resItem[1];
+
+  std::map<std::string, std::vector<Rest::Resources::Body::ListKeyValue>> keyValues;
+  for (auto &key : keys) {
+    keyValues[key.list_element].push_back(key);
+  }
+
+  if (resType == Rest::Resources::Endpoint::ResourceType::Service) { /* If it regards the full service */
+    cubesConfig_.erase(cubeName);
+  } else { /* If it regards an element of that service */
+    nlohmann::json *item = &cubesConfig_[cubeName];
+    bool deleted = false;
+    for (int i = 2; i < resItem.size() - 1; i++) {
+      auto *reference = &(*item)[resItem[i]];
+      if (reference->is_array()) { /* If it is an element of an array, treat it differently */
+        int actualIndex = i;
+        auto toDelete = std::find_if(reference->begin(), reference->end(),
+                [resItem, i, keyValues, this](auto const& elem){return checkPresence(resItem, i, keyValues, elem);});
+        i += keyValues.at(resItem[i]).size();
+        if (i > resItem.size() - 2) { /* if it is an array element, delete it now and set deleted true */
+          reference->erase(toDelete);
+          deleted = true;
+          if (reference->empty()) { /* delete if array is empty */
+            item->erase(resItem[actualIndex]);
+          }
+        } else { /* update the reference to the array elem */
+          reference = &*toDelete;
+        }
+      }
+      item = reference;
+    }
+
+    if (!deleted) {
+      item->erase(resItem[resItem.size() - 1]);
+    }
+  }
+}
+
+void CubesDump::SaveToFile(const std::string& path) {
+  while (true) {
+    // mutex with condition variable waitForUpdate on cubesConfig_
+    std::unique_lock<std::mutex> cubesConfigLock(cubesConfigMutex_);
+    // if there are no updates from last write to file,
+    // either wait for updates (kill=false) or exit if the daemon is shutting down (kill=true)
+    if (toSave_.load() == 0 && !kill_) {
```

Comment:
1) Not clear to me what 'toSave' means.
Is something like "listOfPendingChanges"? In general, select meaningful names.
2) What does "toSave_.load()" means? Why load? Are we loading or saving? The comment is not so helpful either.